tests/ballerina-test/src/test/java/org/ballerinalang/test/connectors/sql/SQLActionsTest.java

```diff
@@ -648,6 +654,416 @@ public void testLoadToMemorySelectAfterTableClose() throws Exception {
                 + "closed table\", cause:null})");
     }
 
+    @Test(groups = "ConnectorTest",
+          description = "Test mapping to nillable type fields")
+    public void testMappingToNillableTypeFields() {
+        BValue[] returns = BRunUtil.invoke(result, "testMappingToNillableTypeFields");
+        Assert.assertNotNull(returns);
+
+        long intExpected = 10;
+        long longExpected = 9223372036854774807L;
+        double floatExpected = 123.34;
+        double doubleExpected = 2139095039;
+        boolean booleanExpected = true;
+        String stringExpected = "Hello";
+        double numericExpected = 1234.567;
+        double decimalExpected = 1234.567;
+        double realExpected = 1234.567;
+        long tinyIntExpected = 1;
+        long smallIntExpected = 5555;
+        String clobExpected = "very long text";
+        String blobExpected = "wso2 ballerina blob test.";
+        String binaryExpected = "wso2 ballerina binary test.";
+        String dateExpected = "{time:1486060200000, zone:{zoneId:\"UTC\", zoneOffset:0}}";
+        String dateTimeExpected = "{time:41745000, zone:{zoneId:\"UTC\", zoneOffset:0}}";
+        String timeExpected = "{time:1486122780000, zone:{zoneId:\"UTC\", zoneOffset:0}}";
+        String timeStampExpected = "{time:1486122780000, zone:{zoneId:\"UTC\", zoneOffset:0}}";
```

Comment:
Shall we directly check for the expected values rather than defining separate variables, It will reduce the code a lot and improves readability